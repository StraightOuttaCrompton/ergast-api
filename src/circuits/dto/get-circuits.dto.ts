import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";

export class GetCircuitsDto {
    /**
     * TODO: extract into class which this inherits from
     */
    @IsOptional()
    @IsInt()
    @IsPositive()
    offset: number;

    @IsOptional()
    @IsInt()
    @IsPositive()
    limit: number;
    /**
     * --------------------------------------------------
     */

    @IsOptional()
    @IsNotEmpty()
    driverId: string;

    @IsOptional()
    @IsNotEmpty()
    constructorId: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status: number; // TODO: validate for actual status values
}
