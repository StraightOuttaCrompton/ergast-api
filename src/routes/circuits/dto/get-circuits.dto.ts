import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetCircuitsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    driverId: string;

    @IsOptional()
    @IsNotEmpty()
    constructorId: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status: number; // TODO: validate for actual status values
}
