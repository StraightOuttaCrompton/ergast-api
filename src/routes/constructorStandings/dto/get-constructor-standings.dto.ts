import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetConstructorStandingsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    constructorId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    position?: number;
}
