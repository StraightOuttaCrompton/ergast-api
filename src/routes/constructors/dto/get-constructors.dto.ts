import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetConstructorsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    circuitId?: string;

    @IsOptional()
    @IsNotEmpty()
    driverId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status?: number; // TODO: validate for actual status values

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    constructorStandings?: number;
}
