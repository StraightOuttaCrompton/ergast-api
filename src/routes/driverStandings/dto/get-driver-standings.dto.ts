import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetDriverStandingsDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    driverId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    position?: number;
}
