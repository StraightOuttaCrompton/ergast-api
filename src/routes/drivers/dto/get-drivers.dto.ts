import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetDriversDto extends CommonDto {
    @IsOptional()
    @IsNotEmpty()
    circuitId?: string;

    @IsOptional()
    @IsNotEmpty()
    constructorId?: string;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    year?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    round?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    grid?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    result?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    fastest?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    status?: number;

    @Type(() => Number)
    @IsOptional()
    @IsInt()
    @IsPositive()
    driverStandings?: number;
}
