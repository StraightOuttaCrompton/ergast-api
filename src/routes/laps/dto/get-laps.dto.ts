import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetLapsDto extends CommonDto {
    @Type(() => Number)
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsInt()
    @IsPositive()
    round: number;

    @IsOptional()
    @IsNotEmpty()
    driverId?: string;
}
