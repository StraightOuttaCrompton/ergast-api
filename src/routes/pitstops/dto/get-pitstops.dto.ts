import { Type } from "class-transformer";
import { IsInt, IsNotEmpty, IsOptional, IsPositive } from "class-validator";
import { CommonDto } from "../../../common.dto";

export class GetPitstopsDto extends CommonDto {
    @Type(() => Number)
    @IsInt()
    @IsPositive()
    year: number;

    @Type(() => Number)
    @IsInt()
    @IsPositive()
    round: number;

    @IsOptional()
    @IsNotEmpty()
    driverId: string;

    @IsOptional()
    @Type(() => Number)
    @IsInt()
    @IsPositive()
    lapNumber: number;
}
