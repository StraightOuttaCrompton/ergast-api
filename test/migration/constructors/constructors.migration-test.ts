import querystring from "querystring";
import { GetConstructorsDto } from "src/routes/constructors/dto/get-constructors.dto";
import { getMigrationTest } from "../migrationUtils";
import path from "path";

describe("GET /constructors/${constructorId} smoke tests", () => {
    const migrationTest = getMigrationTest(
        __dirname,
        path.basename(__filename),
        (response) => response.MRData.ConstructorTable.Constructors[0]
    );

    migrationTest("/constructors/red_bull", `https://ergast.com/api/f1/constructors/red_bull.json`);
});

describe("GET /constructors smoke tests", () => {
    const migrationTest = getMigrationTest(
        __dirname,
        path.basename(__filename),
        (response) => response.MRData.ConstructorTable.Constructors
    );

    const endpoint = "/constructors";

    const queries: { queryParams: Partial<GetConstructorsDto>; legacyRoute: string }[] = [
        {
            queryParams: { driverId: "bottas" },
            legacyRoute: "drivers/bottas",
        },
        {
            queryParams: { driverId: "alonso", circuitId: "monza" },
            legacyRoute: "drivers/alonso/circuits/monza",
        },
        {
            queryParams: { year: 2022 },
            legacyRoute: "2022",
        },
        {
            queryParams: { driverId: "hamilton", year: 2022 },
            legacyRoute: "2022/drivers/hamilton",
        },
        {
            queryParams: { year: 2010, round: 2 },
            legacyRoute: "2010/2",
        },
        {
            queryParams: { driverId: "maldonado", status: 4 },
            legacyRoute: "drivers/maldonado/status/4",
        },
        {
            queryParams: { driverId: "vettel", circuitId: "bahrain", result: 1 },
            legacyRoute: "drivers/vettel/circuits/bahrain/results/1",
        },
        {
            queryParams: { driverId: "hamilton", result: 2, year: 2022 },
            legacyRoute: "2022/drivers/hamilton/results/2",
        },
        {
            queryParams: { driverId: "max_verstappen", grid: 1, year: 2022 },
            legacyRoute: "2022/drivers/max_verstappen/grid/1",
        },
        {
            queryParams: { driverId: "sainz", fastest: 1, year: 2022 },
            legacyRoute: "2022/drivers/sainz/fastest/1",
        },
        {
            queryParams: { constructorStandings: 1, year: 2021 },
            legacyRoute: "2021/constructorStandings/1",
        },
    ];

    migrationTest(endpoint, `https://ergast.com/api/f1/constructors.json`);

    queries.forEach(({ queryParams, legacyRoute }) => {
        migrationTest(
            `${endpoint}?${querystring.stringify(queryParams)}`,
            `https://ergast.com/api/f1/${legacyRoute}/constructors.json`
        );
    });
});
