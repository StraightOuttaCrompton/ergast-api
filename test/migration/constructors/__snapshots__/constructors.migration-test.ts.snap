// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /constructors smoke tests /constructors 1`] = `
[
  {
    "constructorId": "adams",
    "name": "Adams",
    "nationality": "American",
    "url": "http://en.wikipedia.org/wiki/Adams_(constructor)",
  },
  {
    "constructorId": "afm",
    "name": "AFM",
    "nationality": "German",
    "url": "http://en.wikipedia.org/wiki/Alex_von_Falkenhausen_Motorenbau",
  },
  {
    "constructorId": "ags",
    "name": "AGS",
    "nationality": "French",
    "url": "http://en.wikipedia.org/wiki/Automobiles_Gonfaronnaises_Sportives",
  },
  {
    "constructorId": "alfa",
    "name": "Alfa Romeo",
    "nationality": "Swiss",
    "url": "http://en.wikipedia.org/wiki/Alfa_Romeo_in_Formula_One",
  },
  {
    "constructorId": "alphatauri",
    "name": "AlphaTauri",
    "nationality": "Italian",
    "url": "http://en.wikipedia.org/wiki/Scuderia_AlphaTauri",
  },
  {
    "constructorId": "alpine",
    "name": "Alpine F1 Team",
    "nationality": "French",
    "url": "http://en.wikipedia.org/wiki/Alpine_F1_Team",
  },
  {
    "constructorId": "alta",
    "name": "Alta",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/Alta_auto_racing_team",
  },
  {
    "constructorId": "amon",
    "name": "Amon",
    "nationality": "New Zealander",
    "url": "http://en.wikipedia.org/wiki/Amon_(Formula_One_team)",
  },
  {
    "constructorId": "moda",
    "name": "Andrea Moda",
    "nationality": "Italian",
    "url": "http://en.wikipedia.org/wiki/Andrea_Moda_Formula",
  },
  {
    "constructorId": "apollon",
    "name": "Apollon",
    "nationality": "Swiss",
    "url": "http://en.wikipedia.org/wiki/Apollon_(Formula_One)",
  },
  {
    "constructorId": "arrows",
    "name": "Arrows",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/Arrows_Grand_Prix_International",
  },
  {
    "constructorId": "arzani-volpini",
    "name": "Arzani-Volpini",
    "nationality": "Italian",
    "url": "http://en.wikipedia.org/wiki/Arzani-Volpini",
  },
  {
    "constructorId": "butterworth",
    "name": "Aston Butterworth",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/Aston_Butterworth",
  },
  {
    "constructorId": "aston_martin",
    "name": "Aston Martin",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/Aston_Martin_in_Formula_One",
  },
  {
    "constructorId": "ats",
    "name": "ATS",
    "nationality": "Italian",
    "url": "http://en.wikipedia.org/wiki/ATS_(wheels)",
  },
  {
    "constructorId": "bar",
    "name": "BAR",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/British_American_Racing",
  },
  {
    "constructorId": "behra-porsche",
    "name": "Behra-Porsche",
    "nationality": "Italian",
    "url": "http://en.wikipedia.org/wiki/Behra-Porsche",
  },
  {
    "constructorId": "bellasi",
    "name": "Bellasi",
    "nationality": "Swiss",
    "url": "http://en.wikipedia.org/wiki/Bellasi",
  },
  {
    "constructorId": "benetton",
    "name": "Benetton",
    "nationality": "Italian",
    "url": "http://en.wikipedia.org/wiki/Benetton_Formula",
  },
  {
    "constructorId": "bmw",
    "name": "BMW",
    "nationality": "German",
    "url": "http://en.wikipedia.org/wiki/BMW",
  },
  {
    "constructorId": "bmw_sauber",
    "name": "BMW Sauber",
    "nationality": "German",
    "url": "http://en.wikipedia.org/wiki/BMW_Sauber",
  },
  {
    "constructorId": "boro",
    "name": "Boro",
    "nationality": "Dutch",
    "url": "http://en.wikipedia.org/wiki/Boro_(Formula_One)",
  },
  {
    "constructorId": "brabham",
    "name": "Brabham",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/Brabham",
  },
  {
    "constructorId": "brabham-alfa_romeo",
    "name": "Brabham-Alfa Romeo",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/Brabham",
  },
  {
    "constructorId": "brabham-brm",
    "name": "Brabham-BRM",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/Brabham",
  },
  {
    "constructorId": "brabham-climax",
    "name": "Brabham-Climax",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/Brabham",
  },
  {
    "constructorId": "brabham-ford",
    "name": "Brabham-Ford",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/Brabham",
  },
  {
    "constructorId": "brabham-repco",
    "name": "Brabham-Repco",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/Brabham",
  },
  {
    "constructorId": "brawn",
    "name": "Brawn",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/Brawn_GP",
  },
  {
    "constructorId": "brm",
    "name": "BRM",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/BRM",
  },
]
`;

exports[`GET /constructors smoke tests /constructors?constructorStandings=1&year=2021 1`] = `
[
  {
    "constructorId": "mercedes",
    "name": "Mercedes",
    "nationality": "German",
    "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
  },
]
`;

exports[`GET /constructors smoke tests /constructors?driverId=alonso&circuitId=monza 1`] = `
[
  {
    "constructorId": "alpine",
    "name": "Alpine F1 Team",
    "nationality": "French",
    "url": "http://en.wikipedia.org/wiki/Alpine_F1_Team",
  },
  {
    "constructorId": "ferrari",
    "name": "Ferrari",
    "nationality": "Italian",
    "url": "http://en.wikipedia.org/wiki/Scuderia_Ferrari",
  },
  {
    "constructorId": "mclaren",
    "name": "McLaren",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/McLaren",
  },
  {
    "constructorId": "minardi",
    "name": "Minardi",
    "nationality": "Italian",
    "url": "http://en.wikipedia.org/wiki/Minardi",
  },
  {
    "constructorId": "renault",
    "name": "Renault",
    "nationality": "French",
    "url": "http://en.wikipedia.org/wiki/Renault_in_Formula_One",
  },
]
`;

exports[`GET /constructors smoke tests /constructors?driverId=bottas 1`] = `
[
  {
    "constructorId": "alfa",
    "name": "Alfa Romeo",
    "nationality": "Swiss",
    "url": "http://en.wikipedia.org/wiki/Alfa_Romeo_in_Formula_One",
  },
  {
    "constructorId": "mercedes",
    "name": "Mercedes",
    "nationality": "German",
    "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
  },
  {
    "constructorId": "williams",
    "name": "Williams",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/Williams_Grand_Prix_Engineering",
  },
]
`;

exports[`GET /constructors smoke tests /constructors?driverId=hamilton&result=2&year=2022 1`] = `
[
  {
    "constructorId": "mercedes",
    "name": "Mercedes",
    "nationality": "German",
    "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
  },
]
`;

exports[`GET /constructors smoke tests /constructors?driverId=hamilton&year=2022 1`] = `
[
  {
    "constructorId": "mercedes",
    "name": "Mercedes",
    "nationality": "German",
    "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
  },
]
`;

exports[`GET /constructors smoke tests /constructors?driverId=maldonado&status=4 1`] = `
[
  {
    "constructorId": "lotus_f1",
    "name": "Lotus F1",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/Lotus_F1",
  },
  {
    "constructorId": "williams",
    "name": "Williams",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/Williams_Grand_Prix_Engineering",
  },
]
`;

exports[`GET /constructors smoke tests /constructors?driverId=max_verstappen&grid=1&year=2022 1`] = `
[
  {
    "constructorId": "red_bull",
    "name": "Red Bull",
    "nationality": "Austrian",
    "url": "http://en.wikipedia.org/wiki/Red_Bull_Racing",
  },
]
`;

exports[`GET /constructors smoke tests /constructors?driverId=sainz&fastest=1&year=2022 1`] = `
[
  {
    "constructorId": "ferrari",
    "name": "Ferrari",
    "nationality": "Italian",
    "url": "http://en.wikipedia.org/wiki/Scuderia_Ferrari",
  },
]
`;

exports[`GET /constructors smoke tests /constructors?driverId=vettel&circuitId=bahrain&result=1 1`] = `
[
  {
    "constructorId": "ferrari",
    "name": "Ferrari",
    "nationality": "Italian",
    "url": "http://en.wikipedia.org/wiki/Scuderia_Ferrari",
  },
  {
    "constructorId": "red_bull",
    "name": "Red Bull",
    "nationality": "Austrian",
    "url": "http://en.wikipedia.org/wiki/Red_Bull_Racing",
  },
]
`;

exports[`GET /constructors smoke tests /constructors?year=2010&round=2 1`] = `
[
  {
    "constructorId": "ferrari",
    "name": "Ferrari",
    "nationality": "Italian",
    "url": "http://en.wikipedia.org/wiki/Scuderia_Ferrari",
  },
  {
    "constructorId": "force_india",
    "name": "Force India",
    "nationality": "Indian",
    "url": "http://en.wikipedia.org/wiki/Racing_Point_Force_India",
  },
  {
    "constructorId": "hrt",
    "name": "HRT",
    "nationality": "Spanish",
    "url": "http://en.wikipedia.org/wiki/Hispania_Racing",
  },
  {
    "constructorId": "lotus_racing",
    "name": "Lotus",
    "nationality": "Malaysian",
    "url": "http://en.wikipedia.org/wiki/Lotus_Racing",
  },
  {
    "constructorId": "mclaren",
    "name": "McLaren",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/McLaren",
  },
  {
    "constructorId": "mercedes",
    "name": "Mercedes",
    "nationality": "German",
    "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
  },
  {
    "constructorId": "red_bull",
    "name": "Red Bull",
    "nationality": "Austrian",
    "url": "http://en.wikipedia.org/wiki/Red_Bull_Racing",
  },
  {
    "constructorId": "renault",
    "name": "Renault",
    "nationality": "French",
    "url": "http://en.wikipedia.org/wiki/Renault_in_Formula_One",
  },
  {
    "constructorId": "sauber",
    "name": "Sauber",
    "nationality": "Swiss",
    "url": "http://en.wikipedia.org/wiki/Sauber",
  },
  {
    "constructorId": "toro_rosso",
    "name": "Toro Rosso",
    "nationality": "Italian",
    "url": "http://en.wikipedia.org/wiki/Scuderia_Toro_Rosso",
  },
  {
    "constructorId": "virgin",
    "name": "Virgin",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/Virgin_Racing",
  },
  {
    "constructorId": "williams",
    "name": "Williams",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/Williams_Grand_Prix_Engineering",
  },
]
`;

exports[`GET /constructors smoke tests /constructors?year=2022 1`] = `
[
  {
    "constructorId": "alfa",
    "name": "Alfa Romeo",
    "nationality": "Swiss",
    "url": "http://en.wikipedia.org/wiki/Alfa_Romeo_in_Formula_One",
  },
  {
    "constructorId": "alphatauri",
    "name": "AlphaTauri",
    "nationality": "Italian",
    "url": "http://en.wikipedia.org/wiki/Scuderia_AlphaTauri",
  },
  {
    "constructorId": "alpine",
    "name": "Alpine F1 Team",
    "nationality": "French",
    "url": "http://en.wikipedia.org/wiki/Alpine_F1_Team",
  },
  {
    "constructorId": "aston_martin",
    "name": "Aston Martin",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/Aston_Martin_in_Formula_One",
  },
  {
    "constructorId": "ferrari",
    "name": "Ferrari",
    "nationality": "Italian",
    "url": "http://en.wikipedia.org/wiki/Scuderia_Ferrari",
  },
  {
    "constructorId": "haas",
    "name": "Haas F1 Team",
    "nationality": "American",
    "url": "http://en.wikipedia.org/wiki/Haas_F1_Team",
  },
  {
    "constructorId": "mclaren",
    "name": "McLaren",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/McLaren",
  },
  {
    "constructorId": "mercedes",
    "name": "Mercedes",
    "nationality": "German",
    "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
  },
  {
    "constructorId": "red_bull",
    "name": "Red Bull",
    "nationality": "Austrian",
    "url": "http://en.wikipedia.org/wiki/Red_Bull_Racing",
  },
  {
    "constructorId": "williams",
    "name": "Williams",
    "nationality": "British",
    "url": "http://en.wikipedia.org/wiki/Williams_Grand_Prix_Engineering",
  },
]
`;

exports[`GET /constructors/\${constructorId} smoke tests /constructors/red_bull 1`] = `
{
  "constructorId": "red_bull",
  "name": "Red Bull",
  "nationality": "Austrian",
  "url": "http://en.wikipedia.org/wiki/Red_Bull_Racing",
}
`;
